File: vb.net2010/CameraControl/Form1.vb
Public Class Form1

    Private Sub Form1_Load(sender As System.Object, e As System.EventArgs) Handles MyBase.Load
        Dim i As Integer


        For i = 0 To AxVideoCap1.GetVideoInputCount - 1
            cboVideoInput.Items.Add(AxVideoCap1.GetVideoInputName(i))
        Next

        If cboVideoInput.Items.Count > 0 Then
            cboVideoInput.SelectedIndex = 0
        End If

        For i = 0 To AxVideoCap1.GetVideoFormatCount - 1
            cbovideoformat.Items.Add(AxVideoCap1.GetVideoFormatName(i))
        Next

        If cbovideoformat.Items.Count > 0 Then
            cbovideoformat.SelectedIndex = 0
        End If
    End Sub

    Private Sub Command1_Click(sender As System.Object, e As System.EventArgs) Handles Command1.Click
        AxVideoCap1.VideoInput = cboVideoInput.SelectedIndex
        AxVideoCap1.VideoFormat = cbovideoformat.SelectedIndex

        Me.AxVideoCap1.Start()

        cmdProPage.Enabled = True

        If Me.AxVideoCap1.CameraFocus.Available Then

            If AxVideoCap1.CameraFocus.Auto Then
                chkfocusauto.Checked = True
            Else
                chkfocusauto.Checked = False
            End If

            FocusTrackBar.Minimum = AxVideoCap1.CameraFocus.Min
            FocusTrackBar.Maximum = AxVideoCap1.CameraFocus.Max
            FocusTrackBar.Value = AxVideoCap1.CameraFocus.Value
        Else
            FocusTrackBar.Enabled = False

        End If


        If Me.AxVideoCap1.Brightness.Available Then
            Me.BrightnessSlider1.Minimum = Me.AxVideoCap1.Brightness.Min
            Me.BrightnessSlider1.Maximum = Me.AxVideoCap1.Brightness.Max
            Me.BrightnessSlider1.Value = Me.AxVideoCap1.Brightness.Value
        Else
            Me.BrightnessSlider1.Enabled = False
        End If



        If Me.AxVideoCap1.Contrast.Available Then
            Me.ContrastSlider1.Maximum = Me.AxVideoCap1.Contrast.Min
            Me.ContrastSlider1.Maximum = Me.AxVideoCap1.Contrast.Max
            Me.ContrastSlider1.Value = Me.AxVideoCap1.Contrast.Value
        Else
            Me.ContrastSlider1.Enabled = False
        End If

        If Me.AxVideoCap1.Hue.Available Then
            Me.HueSlider1.Minimum = Me.AxVideoCap1.Hue.Min
            Me.HueSlider1.Maximum = Me.AxVideoCap1.Hue.Max
            Me.HueSlider1.Value = Me.AxVideoCap1.Hue.Value
        Else
            Me.HueSlider1.Enabled = False
        End If

        If Me.AxVideoCap1.Saturation.Available Then
            Me.SaturationSlider1.Minimum = Me.AxVideoCap1.Saturation.Min
            Me.SaturationSlider1.Maximum = Me.AxVideoCap1.Saturation.Max
            Me.SaturationSlider1.Value = Me.AxVideoCap1.Saturation.Value
        Else
            Me.SaturationSlider1.Enabled = False
        End If


        If Me.AxVideoCap1.Sharpness.Available Then
            Me.SharpnessSlider1.Minimum = Me.AxVideoCap1.Sharpness.Min
            Me.SharpnessSlider1.Maximum = Me.AxVideoCap1.Sharpness.Max
            Me.SharpnessSlider1.Value = Me.AxVideoCap1.Sharpness.Value
        Else
            Me.SharpnessSlider1.Enabled = False
        End If


        If Me.AxVideoCap1.Gamma.Available Then
            Me.GammaSlider1.Minimum = Me.AxVideoCap1.Gamma.Min
            Me.GammaSlider1.Maximum = Me.AxVideoCap1.Gamma.Max
            Me.GammaSlider1.Value = Me.AxVideoCap1.Gamma.Value
        Else
            Me.GammaSlider1.Enabled = False
        End If

        If Me.AxVideoCap1.ColorEnable.Available Then
            Me.ColorEnableSlider1.Minimum = Me.AxVideoCap1.ColorEnable.Min
            Me.ColorEnableSlider1.Maximum = Me.AxVideoCap1.ColorEnable.Max
            Me.ColorEnableSlider1.Value = Me.AxVideoCap1.ColorEnable.Value
        Else
            Me.ColorEnableSlider1.Enabled = False
        End If

        If Me.AxVideoCap1.WhiteBalance.Available Then

            If AxVideoCap1.WhiteBalance.Auto Then
                chkwhitebalauto.Checked = True
            Else
                chkwhitebalauto.Checked = False
            End If

            Me.WhiteBalanceSlider1.Minimum = Me.AxVideoCap1.WhiteBalance.Min
            Me.WhiteBalanceSlider1.Maximum = Me.AxVideoCap1.WhiteBalance.Max
            Me.WhiteBalanceSlider1.Value = Me.AxVideoCap1.WhiteBalance.Value
        Else
            Me.WhiteBalanceSlider1.Enabled = False
        End If

        If Me.AxVideoCap1.BacklightCompensation.Available Then
            Me.BacklightCompensationSlider1.Minimum = Me.AxVideoCap1.BacklightCompensation.Min
            Me.BacklightCompensationSlider1.Maximum = Me.AxVideoCap1.BacklightCompensation.Max
            Me.BacklightCompensationSlider1.Value = Me.AxVideoCap1.BacklightCompensation.Value
        Else
            Me.BacklightCompensationSlider1.Enabled = False
        End If


        If Me.AxVideoCap1.Gain.Available Then
            Me.GainSlider1.Minimum = Me.AxVideoCap1.Gain.Min
            Me.GainSlider1.Maximum = Me.AxVideoCap1.Gain.Max
            Me.GainSlider1.Value = Me.AxVideoCap1.Gain.Value
        Else
            Me.GainSlider1.Enabled = False
        End If


        If Me.AxVideoCap1.Zoom.Available Then
            Me.ZoomSlider1.Minimum = Me.AxVideoCap1.Zoom.Min
            Me.ZoomSlider1.Maximum = Me.AxVideoCap1.Zoom.Max
            Me.ZoomSlider1.Value = Me.AxVideoCap1.Zoom.Value
        Else
            Me.ZoomSlider1.Enabled = False
        End If

    End Sub

    Private Sub cmdProPage_Click(sender As System.Object, e As System.EventArgs) Handles cmdProPage.Click
        Me.AxVideoCap1.ShowVideoCapturePropertyPage(0)
    End Sub

    Private Sub BrightnessSlider1_Scroll(sender As System.Object, e As System.EventArgs) Handles BrightnessSlider1.Scroll
        AxVideoCap1.Brightness().Value = Me.BrightnessSlider1.Value
    End Sub

    Private Sub ContrastSlider1_Scroll(sender As System.Object, e As System.EventArgs) Handles ContrastSlider1.Scroll
        AxVideoCap1.Contrast.Value = Me.ContrastSlider1.Value
    End Sub

    Private Sub HueSlider1_Scroll(sender As System.Object, e As System.EventArgs) Handles HueSlider1.Scroll
        AxVideoCap1.Hue.Value = Me.HueSlider1.Value
    End Sub

    Private Sub SaturationSlider1_Scroll(sender As System.Object, e As System.EventArgs) Handles SaturationSlider1.Scroll
        AxVideoCap1.Saturation.Value = Me.SaturationSlider1.Value
    End Sub

    Private Sub SharpnessSlider1_Scroll(sender As System.Object, e As System.EventArgs) Handles SharpnessSlider1.Scroll
        AxVideoCap1.Sharpness.Value = Me.SharpnessSlider1.Value
    End Sub

    Private Sub GammaSlider1_Scroll(sender As System.Object, e As System.EventArgs) Handles GammaSlider1.Scroll
        AxVideoCap1.Gamma.Value = Me.GammaSlider1.Value
    End Sub

    Private Sub ColorEnableSlider1_Scroll(sender As System.Object, e As System.EventArgs) Handles ColorEnableSlider1.Scroll
        AxVideoCap1.ColorEnable.Value = Me.ColorEnableSlider1.Value
    End Sub

    Private Sub WhiteBalanceSlider1_Scroll(sender As System.Object, e As System.EventArgs) Handles WhiteBalanceSlider1.Scroll
        AxVideoCap1.WhiteBalance.Value = Me.WhiteBalanceSlider1.Value
    End Sub

    Private Sub BacklightCompensationSlider1_Scroll(sender As System.Object, e As System.EventArgs) Handles BacklightCompensationSlider1.Scroll
        AxVideoCap1.BacklightCompensation.Value = Me.BacklightCompensationSlider1.Value
    End Sub

    Private Sub GainSlider1_Scroll(sender As System.Object, e As System.EventArgs) Handles GainSlider1.Scroll
        AxVideoCap1.Gain.Value = Me.GainSlider1.Value
    End Sub

    Private Sub ZoomSlider1_Scroll(sender As System.Object, e As System.EventArgs) Handles ZoomSlider1.Scroll
        AxVideoCap1.Zoom.Value = Me.ZoomSlider1.Value
    End Sub

    Private Sub FocusTrackBar_Scroll(sender As System.Object, e As System.EventArgs) Handles FocusTrackBar.Scroll
        AxVideoCap1.CameraFocus.Value = Me.FocusTrackBar.Value
    End Sub

    Private Sub chkfocusauto_CheckedChanged(sender As System.Object, e As System.EventArgs) Handles chkfocusauto.CheckedChanged
        If AxVideoCap1.CameraFocus.Available Then

            If chkfocusauto.Checked Then

                AxVideoCap1.CameraFocus.Auto = True
                FocusTrackBar.Enabled = False

            Else

                AxVideoCap1.CameraFocus.Auto = False
                FocusTrackBar.Enabled = True

            End If
        End If
    End Sub

    Private Sub chkwhitebalauto_CheckedChanged(sender As System.Object, e As System.EventArgs) Handles chkwhitebalauto.CheckedChanged
        If AxVideoCap1.WhiteBalance.Available Then


            If chkwhitebalauto.Checked Then

                AxVideoCap1.WhiteBalance.Auto = True
                WhiteBalanceSlider1.Enabled = False

            Else

                AxVideoCap1.WhiteBalance.Auto = False
                WhiteBalanceSlider1.Enabled = True



            End If
        End If
    End Sub
End Class

File: c#2010/CameraControl/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace WindowsFormsApplication1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void previewButton_Click(object sender, EventArgs e)
        {
            this.axVideoCap1.Device = 0;
            this.axVideoCap1.VideoInput = (short)this.videoInputComboBox.SelectedIndex;

            this.axVideoCap1.Start();
            this.proPageButton.Enabled = true;

            if (this.axVideoCap1.CameraFocus.Available)
            {
                if (axVideoCap1.CameraFocus.Auto)
                    chkfocusauto.Checked = true;
                else
                    chkfocusauto.Checked = false;


                this.FocusTrackBar.Minimum = this.axVideoCap1.CameraFocus.Min;
                this.FocusTrackBar.Maximum = this.axVideoCap1.CameraFocus.Max;
                this.FocusTrackBar.Value = this.axVideoCap1.CameraFocus.Value;

            }
            else
                this.FocusTrackBar.Enabled = false;

            if (this.axVideoCap1.Brightness.Available)
            {
                this.brightnessTrackBar.Minimum = this.axVideoCap1.Brightness.Min;
                this.brightnessTrackBar.Maximum = this.axVideoCap1.Brightness.Max;
                this.brightnessTrackBar.Value = this.axVideoCap1.Brightness.Value;
            }
            else
            {
                this.brightnessTrackBar.Enabled = false;
            }



            if (this.axVideoCap1.Contrast.Available)
            {
                this.contrastTrackBar.Minimum = this.axVideoCap1.Contrast.Min;
                this.contrastTrackBar.Maximum = this.axVideoCap1.Contrast.Max;
                this.contrastTrackBar.Value = this.axVideoCap1.Contrast.Value;
            }
            else
            {
                this.contrastTrackBar.Enabled = false;
            }

            if (this.axVideoCap1.Hue.Available)
            {
                this.hueTrackBar.Minimum = this.axVideoCap1.Hue.Min;
                this.hueTrackBar.Maximum = this.axVideoCap1.Hue.Max;
                this.hueTrackBar.Value = this.axVideoCap1.Hue.Value;
            }
            else
            {
                this.hueTrackBar.Enabled = false;
            }

            if (this.axVideoCap1.Saturation.Available)
            {
                this.saturationTrackBar.Minimum = this.axVideoCap1.Saturation.Min;
                this.saturationTrackBar.Maximum = this.axVideoCap1.Saturation.Max;
                this.saturationTrackBar.Value = this.axVideoCap1.Saturation.Value;
            }
            else
            {
                this.saturationTrackBar.Enabled = false;
            }


            if (this.axVideoCap1.Sharpness.Available)
            {
                this.sharpnessTrackBar.Minimum = this.axVideoCap1.Sharpness.Min;
                this.sharpnessTrackBar.Maximum = this.axVideoCap1.Sharpness.Max;
                this.sharpnessTrackBar.Value = this.axVideoCap1.Sharpness.Value;
            }
            else
            {
                this.sharpnessTrackBar.Enabled = false;
            }


            if (this.axVideoCap1.Gamma.Available)
            {
                this.gammaTrackBar.Minimum = this.axVideoCap1.Gamma.Min;
                this.gammaTrackBar.Maximum = this.axVideoCap1.Gamma.Max;
                this.gammaTrackBar.Value = this.axVideoCap1.Gamma.Value;
            }
            else
            {
                this.gammaTrackBar.Enabled = false;
            }

            if (this.axVideoCap1.ColorEnable.Available)
            {
                this.colorEnableTrackBar.Minimum = this.axVideoCap1.ColorEnable.Min;
                this.colorEnableTrackBar.Maximum = this.axVideoCap1.ColorEnable.Max;
                this.colorEnableTrackBar.Value = this.axVideoCap1.ColorEnable.Value;
            }
            else
            {
                this.colorEnableTrackBar.Enabled = false;
            }

            if (this.axVideoCap1.WhiteBalance.Available)
            {
                if (this.axVideoCap1.WhiteBalance.Auto)
                    chkwhitebalauto.Checked = true;
                else
                    chkwhitebalauto.Checked = false;

                this.whiteBalanceTrackBar.Minimum = this.axVideoCap1.WhiteBalance.Min;
                this.whiteBalanceTrackBar.Maximum = this.axVideoCap1.WhiteBalance.Max;
                this.whiteBalanceTrackBar.Value = this.axVideoCap1.WhiteBalance.Value;
            }
            else
            {
                this.whiteBalanceTrackBar.Enabled = false;
            }

            if (this.axVideoCap1.BacklightCompensation.Available)
            {
                this.backlightCompensationTrackBar.Minimum = this.axVideoCap1.BacklightCompensation.Min;
                this.backlightCompensationTrackBar.Maximum = this.axVideoCap1.BacklightCompensation.Max;
                this.backlightCompensationTrackBar.Value = this.axVideoCap1.BacklightCompensation.Value;
            }
            else
            {
                this.backlightCompensationTrackBar.Enabled = false;
            }


            if (this.axVideoCap1.Gain.Available)
            {
                this.gainTrackBar.Minimum = this.axVideoCap1.Gain.Min;
                this.gainTrackBar.Maximum = this.axVideoCap1.Gain.Max;
                this.gainTrackBar.Value = this.axVideoCap1.Gain.Value;
            }
            else
            {
                this.gainTrackBar.Enabled = false;
            }
        }

        private void proPageButton_Click(object sender, EventArgs e)
        {
            this.axVideoCap1.ShowVideoCapturePropertyPage(0);
        }

        private void brightnessTrackBar_Scroll(object sender, EventArgs e)
        {
            this.axVideoCap1.Brightness.Value = this.brightnessTrackBar.Value;
        }

        private void contrastTrackBar_Scroll(object sender, EventArgs e)
        {
            this.axVideoCap1.Contrast.Value = this.contrastTrackBar.Value;
        }

        private void hueTrackBar_Scroll(object sender, EventArgs e)
        {
            this.axVideoCap1.Hue.Value = this.hueTrackBar.Value;
        }

        private void FocusTrackBar_Scroll(object sender, EventArgs e)
        {
            this.axVideoCap1.CameraFocus.Value = this.FocusTrackBar.Value;
        }

        private void saturationTrackBar_Scroll(object sender, EventArgs e)
        {
            this.axVideoCap1.Saturation.Value = this.hueTrackBar.Value;
        }

        private void sharpnessTrackBar_Scroll(object sender, EventArgs e)
        {
            this.axVideoCap1.Sharpness.Value = this.sharpnessTrackBar.Value;
        }

        private void gammaTrackBar_Scroll(object sender, EventArgs e)
        {
            this.axVideoCap1.Gamma.Value = this.gainTrackBar.Value;
        }

        private void colorEnableTrackBar_Scroll(object sender, EventArgs e)
        {
            this.axVideoCap1.ColorEnable.Value = this.colorEnableTrackBar.Value;
        }

        private void whiteBalanceTrackBar_Scroll(object sender, EventArgs e)
        {
            this.axVideoCap1.WhiteBalance.Value = this.whiteBalanceTrackBar.Value;
        }

        private void backlightCompensationTrackBar_Scroll(object sender, EventArgs e)
        {
            this.axVideoCap1.BacklightCompensation.Value = this.backlightCompensationTrackBar.Value;
        }

        private void gainTrackBar_Scroll(object sender, EventArgs e)
        {
            this.axVideoCap1.Gain.Value = this.gainTrackBar.Value;
        }

        private void chkfocusauto_CheckedChanged(object sender, EventArgs e)
        {
            if (this.axVideoCap1.CameraFocus.Available)
            {
                if (chkfocusauto.Checked)
                {
                    this.axVideoCap1.CameraFocus.Auto = true;
                    this.FocusTrackBar.Enabled = false;
                }
                else
                {
                    this.axVideoCap1.CameraFocus.Auto = false;
                    this.FocusTrackBar.Enabled = true;
                }
            }
        }

        private void chkwhitebalauto_CheckedChanged(object sender, EventArgs e)
        {
            if (this.axVideoCap1.WhiteBalance.Available)
            {

                if (chkwhitebalauto.Checked)
                {
                    this.axVideoCap1.WhiteBalance.Auto = true;
                    this.whiteBalanceTrackBar.Enabled = false;
                }
                else
                {
                    this.axVideoCap1.WhiteBalance.Auto = false;
                    this.whiteBalanceTrackBar.Enabled = true;
                }

            }
        }

        private void Form1_Load(object sender, EventArgs e)
        {

        }

        private void btnbrightnessdefault_Click(object sender, EventArgs e)
        {
            axVideoCap1.Brightness.Value = axVideoCap1.Brightness.Default;
        }

        private void btncontrastdefault_Click(object sender, EventArgs e)
        {
            axVideoCap1.Contrast.Value = axVideoCap1.Contrast.Default;
        }

        private void btnhuedefault_Click(object sender, EventArgs e)
        {
            axVideoCap1.Hue.Value = axVideoCap1.Hue.Default;
        }

        private void btnsatdefault_Click(object sender, EventArgs e)
        {
            axVideoCap1.Saturation.Value = axVideoCap1.Saturation.Default;
        }

        private void btnsharpnessdefault_Click(object sender, EventArgs e)
        {
            axVideoCap1.Sharpness.Value = axVideoCap1.Sharpness.Default;
        }

        private void btngammadefault_Click(object sender, EventArgs e)
        {
            axVideoCap1.Gamma.Value = axVideoCap1.Gamma.Default;
        }

        private void btncolordefault_Click(object sender, EventArgs e)
        {
            axVideoCap1.ColorEnable.Value = axVideoCap1.ColorEnable.Default;
        }

        private void btnwhitebalancedefault_Click(object sender, EventArgs e)
        {
            axVideoCap1.WhiteBalance.Value = axVideoCap1.WhiteBalance.Default;
        }

        private void btnbacklightdefault_Click(object sender, EventArgs e)
        {
            axVideoCap1.BacklightCompensation.Value = axVideoCap1.BacklightCompensation.Default;
        }

        private void btngaindefault_Click(object sender, EventArgs e)
        {
            axVideoCap1.Gain.Value = axVideoCap1.Gain.Default;
        }
    }
}
